### PR TITLE
[All] Migrate to AGP 8.0

### PR DESCRIPTION
**Pull request by mlykotom**
- update to Gradle 8.1 and removes DSL_SCOPE_VIOLATION
- updates to AGP 8.0 in all projects
- updates to JAVA_VERSION.VERSION_17
- migrates packagingOptions --> packaging.resources
- updates workflows to use JDK 17 + zulu distribution
- updates workflows to use build-sample.yml from current commit rather than using it from `@main`

### CODE REVIEW

**Comment by mlykotom** [comment]
For some reason, API 29 time outs for JetNews :(

**Comment by astamato** [comment]
@mlykotom rerun and passed.

**Review by astamato — APPROVED** [review]
LGTM

**Review by zsmb13 — APPROVED** [review]
Thanks for cleaning up the Kotlin DLS / version catalog warnings!

File: JetNews/app/build.gradle.kts
Comment (percy-g2) [inline JetNews/app/build.gradle.kts:1]: we can convert put to assignment , if we are modifying this file along the way ? and for the rest of samples as well.
Rest is ACK from me :)

**Pull request merged**